pkg/ast/pipesearch/searchHandler.go

```diff
@@ -593,6 +603,149 @@ func RunQueryForNewPipeline(conn *websocket.Conn, qid uint64, root *structs.ASTN
 	}
 }
 
+func getFinalBuckets(complete *structs.PipeSearchCompleteResponse, timeRange *dtu.TimeRange) error {
+
+	spanOptions, err := structs.GetDefaultTimechartSpanOptions(timeRange.StartEpochMs, timeRange.EndEpochMs, 0)
+	if err != nil {
+		return fmt.Errorf("failed to get default timechart span options: %w", err)
+	}
+
+	interval := spanOptions.SpanLength.Num
+	timeScaler := spanOptions.SpanLength.TimeScalr
+
+	resultMap := make(map[string]*structs.BucketHolder)
```

Comment:
nit: rename this to be `buckHolderMap` or something else, reslutMap is too generic, and the reviewer/reader has to back up in the code to understand what result map it is